react/features/base/settings/reducer.js

```diff
@@ -116,17 +116,24 @@ function _initSettings(featureState) {
     }, settings);
 
     if (!browser.isReactNative()) {
+        const persistedSettings = JSON.parse(
```

Comment:
I guess there is a difference, I just can't see it, so why did you get the feature's persisted values here and merge the fields one-by-one as opposed to letting the fields constructed into an object and get that merged at line with ```assignIfDefined``` below?

Reply:
Good point. The issue here is if the user has an old value for micdeviceid, maybe one that got set before the settings changes. So if I had micA as my preferred, then loaded the js with features/base/settings, and then changed my preferred to micB, because micA still exists micA will be set as preferred instead of micB.

Reply:
For transparency: I have no idea what I was looking at when I was reproing the issue with mic and camera. There is no issue. I have no explanation for what I was seeing other than I was being stupid. Speaker persistence does seem to have an issue on beta/meet though.